### DISCUSSION TITLE
Fatal error: Uncaught Phalcon\Filter\Exception: Filter absint is not registered

### DISCUSSION THREAD

**Discussion by hostbox2** [phalcon/cphalcon#16398] [General]
``From the manual section: https://docs.phalcon.io/5.0/en/filter-filter#methods
I took the code without modification:
```php
<?php

use Phalcon\Filter\Filter;

$filter = new Filter();
$source = -123;

echo $filter->absint($source); // 123
``` 

And ran it. The result was an error:
`Fatal error: Uncaught Phalcon\Filter\Exception: Filter absint is not registered in ...../phalconFilter.php:7`

In fact, I get exactly the same error in the working code as well. But I decided to check it on a standard example. And it doesn't work.
The same error if you enter any other type, such as string, etc.

**Details**
 - Phalcon 5.2.1
 - PHP Version: 8.1.9
 - Operating System: Windows 10 LTSC
 - Server: Apache (OSPanel server)

**Comment by s-ohnishi** [comment]
I think that manual is wrong.
I have looked at [filter.zep](https://github.com/phalcon/cphalcon/blob/master/phalcon/Filter/Filter.zep), but I think that only the services registered at the time of instantiation can be used.

If you use FilterFactory, it will register a known filter, so I think it's good to use it as follows.
```PHP
$source = -123;
$factory = new \Phalcon\Filter\FilterFactory();
$filter = $factory->newInstance();
echo $filter->absint($source); // 123
```

**Comment by niden** [comment]
The manual was indeed wrong (Thank you @s-ohnishi )

I have corrected it: https://docs.phalcon.io/5.0/en/filter-filter#methods

In short, you can use the FilterFactory and get a new filter using `newInstance()` then all the built in sanitizers will be enabled.

**Reply by s-ohnishi** [reply]
@niden, I confirmed.

I find it redundant to create an instance of FilterFactory and then get the filter object.
Probably because I think that $factory (instance of FilterFactory) will not be used for anything else.
I think it would be easier if you could get the $filter (instance of Filter) registered with the service using the FilterFactory static method.

**Reply by niden** [reply]
Yes the filter will need some work in the near future. I want to make it easier for people to use or not use the built in filters and also add their own (right now that comes with an array in the constructor.

I also want to really merge the validators and the filters together and name them accordingly i.e. `absint()` is really a sanitizer where something like `isAbsInt()` would be a validator - something in those lines

That is v6 work that will come to v5 also once I get time to work on it.

**Reply by s-ohnishi** [reply]
When I add my own filter, I do the following when adding the filter service with DI, but is it easier?
``` PHP
$factory  = new \Phalcon\Filter\FilterFactory();
$filter = $factory->newInstance();
$filter->set('mbTrim', function ($str) {
    if ($str === null) {
         return null;
     }
     return preg_replace(
         "/\A[\p{Cc}\p{Zs}]++|[\p{Cc}\p{Zs}]++\z/u",
         '', $str);
});
```

It would be very convenient if we could implement a mechanism to become a validator just by adding is to the filter name (sanitizer name). :-0

**Reply by niden** [reply]
Yes that is exactly how it is done and you can have the built in services along with yours. I would like to make it a bit easier though especially with the factory there.

Another todo....

**Reply by s-ohnishi** [reply]
I am waiting with anticipation. ;-)
